[delishad21] iP

delishad21 opened the pull request:
<h1> CBWBot </h1>

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

CBWBot frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/delishad21/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [ ]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

delishad21 committed:
Implemented startup and exit messages

delishad21 committed:
Added echo functionality

delishad21 committed:
Implemented task adding

delishad21 committed:
Created a new class for Tasks, added functionality for marking and unmarking tasks

delishad21 committed:
Implemented different task types (ToDos, Deadlines, Events) as subclasses of task.

delishad21 committed:
Added error handling and new exceptions

delishad21 committed:
Added task deletetion capabilities

delishad21 committed:
Automated text UI testing

delishad21 committed:
Add data saving and reading save data from file

delishad21 committed:
Merge branch 'branch-Level-7'

delishad21 committed:
Add datetime functionality

delishad21 committed:
Merge branch 'branch-Level-8'

delishad21 committed:
Add classes for OOP structuring

delishad21 committed:
Merge branch 'branch-A-MoreOOP'

delishad21 committed:
Sort classes into packages

delishad21 committed:
Merge branch 'branch-A-Packages'

delishad21 committed:
Merge branch 'add-gradle-support' of https://github.com/delishad21/ip

delishad21 committed:
Add test classes

delishad21 committed:
Merge branch 'branch-A-JUnit'

delishad21 committed:
Stop tracking settings.json and add build info

delishad21 committed:
Add comments for javadoc

delishad21 committed:
Modify code to coding standards

delishad21 committed:
Add find capabilities

delishad21 committed:
Merge branch 'branch-A-JavaDoc'

delishad21 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/commands/ModifyTaskCommand.java
#	src/main/java/duke/tasks/Deadline.java
#	src/main/java/duke/tasks/Event.java

delishad21 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/tasks/Task.java
#	src/main/java/duke/utils/Parser.java

delishad21 committed:
Update javadoc comments

delishad21 committed:
Update files to follow coding standards

delishad21 committed:
Implement basic Javafx interface

delishad21 committed:
Merge branch 'branch-A-CheckStyle'

delishad21 committed:
Add javadoc comments for UI classes and methods

delishad21 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	build.gradle
#	src/main/java/duke/Duke.java
#	src/main/java/duke/commands/Command.java
#	src/main/java/duke/commands/ExitCommand.java
#	src/main/java/duke/commands/GenerateTaskCommand.java
#	src/main/java/duke/commands/ListTaskCommand.java
#	src/main/java/duke/commands/ModifyTaskCommand.java
#	src/main/java/duke/tasks/Todo.java
#	src/main/java/duke/utils/Parser.java
#	src/main/java/duke/utils/Storage.java
#	src/main/java/duke/utils/Ui.java

delishad21 committed:
Improve UI features

delishad21 committed:
Update root path in build.gradle

alvinnzz commented on src/main/java/duke/commands/ModifyTaskCommand.java:
```diff
@@ -0,0 +1,76 @@
+    /**
+     * Enum used for classifying task modification types
+     */
+    public enum ModificationTypes { MARK, UNMARK, DELETE }
```
I like how you used enum to encapsulate these 3 commands.

alvinnzz commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,61 @@
+    public static final DateTimeFormatter OUTPUT_DT_FORMATTER =
+        DateTimeFormatter.ofPattern("dd MMMM yyyy, hh:mma", Locale.ENGLISH);
+
+
```
I think one line of spacing would be fine

alvinnzz commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,61 @@
+            return new ModifyTaskCommand(ModifyTaskCommand.ModificationTypes.DELETE, input);
+        default:
+            throw new NoSuchCommandException(input);
+        }
```
I like how neat your switch case are, adhering to the coding standards.

alvinnzz commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,137 @@
+ * @author delishad21
+ */
+public class Storage {
+    private File f;
```
Perhaps a better name can be used?

DhiraPT replied:
I agree with @alvinnzz. Perhaps the variable name can explain the entity better?

alvinnzz commented on src/main/java/duke/utils/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+ * @author delishad21
+ */
+public class TaskList {
+    private ArrayList<Task> toDoList;
```
Perhaps taskList may be a better name, as toDo sounds like one of the task types?

alvinnzz reviewed (commented):
Generally very neat code, adhering to the coding standards!

delishad21 committed:
Improve code quality by adding new methods and classes

delishad21 committed:
Add assertions to maintain integrity of data at runtime

delishad21 committed:
Merge pull request #2 from delishad21/branch-A-Assertions

Add assertions to maintain integrity of data at runtime

DhiraPT commented on src/test/java/duke/utils/StorageTest.java:
```diff
@@ -0,0 +1,58 @@
+            FileWriter fw = new FileWriter(file);
+            fw.write("[T]|[ ]|1\n" +
+                    "[D]|[X]|task2|12/11/2001, 16:00\n" +
+                    "[E]|[ ]|task 3|12/11/2001, 15:00|13/11/2001, 16:00");
```
Would wrapping the line before the operator `+` be better?

DhiraPT commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+     * @return String
+     */
+    @Override
+    public String toSave() {
```
Perhaps you can use a more descriptive method name. `toSave()` might imply a method to save the list, rather than to return the string representation of `Deadline`. I noticed the same issue in some other classes as well.

DhiraPT commented on src/main/java/duke/tasks/Todo.java:
```diff
@@ -0,0 +1,65 @@
+ */
+public class Todo extends Task {
+
+    private static String REQUIRED_PARAMS[] = {"description"};
```
Any reason you use SCREAMING_SNAKE_CASE here? Typically, this case is reserved for constants (variables that are `static final`). I noticed the same issue in several other classes too.

DhiraPT commented on src/main/java/duke/exceptions/TaskModificationException.java:
```diff
@@ -0,0 +1,18 @@
+     * @param errorMsg Reason for task modification failure.
+     */
+    public TaskModificationException(String errorMsg) {
+        super(errorMsg);
```
Would changing `errorMsg` to `errorMessage` be better for easier pronunciation and readability?

DhiraPT reviewed (commented):
Overall, well done!

delishad21 committed:
Merge branch 'master' into branch-A-CodeQuality

delishad21 committed:
Merge pull request #2 from delishad21/branch-A-CodeQuality

Improve code quality by adding new methods and classes

delishad21 committed:
Update README.md

delishad21 committed:
Add .yml file for Continuous Integration on github

delishad21 committed:
Modify test files to coding standard

delishad21 committed:
Add functionality to save contacts together with tasks

CBBW bot is currently only able to save tasks of different types

The functionality of the bot is limited

Add ability to save contacts to the bot through changing the data recording method

by having a database class to save all records, this opens up more opportunities
to add further functionalities to the bot

delishad21 committed:
Add functionality to save contacts together with tasks

CBBW bot is currently only able to save tasks of different types

The functionality of the bot is limited

Add ability to save contacts to the bot through changing the data recording method

by having a database class to save all records, this opens up more opportunities
to add further functionalities to the bot

delishad21 committed:
Improve UI with new background and chat bubbles

delishad21 committed:
Change names to Grizzly, improve GUI, add personality

delishad21 committed:
Add increased time before application exit to see exit messages

delishad21 committed:
Add formatting changes to text

delishad21 committed:
Add title to main window

delishad21 committed:
Seperate list functionality for Contacts and Tasks

delishad21 committed:
Update README.md for Pages

delishad21 committed:
Create _config.yml for pages

delishad21 committed:
Update _config.yml

delishad21 committed:
Add JUnit testing for most public methods

delishad21 committed:
Fix bug with backgound moving on GUI

delishad21 committed:
Add help functionality

delishad21 committed:
Update links in pages readme
